### PR TITLE
make ApplyG2pJob not a mini_task, since it can be quite long

### PR DESCRIPTION
**Pull request by mattiadg**
This job can take also hours to finish, according to the vocabulary size. 
I think it's better not to have it as a mini_task

### CODE REVIEW

File: g2p/apply.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __init__(
         self.out_g2p_lexicon = self.output_path("g2p.lexicon")
 
     def tasks(self):
-        yield Task("run", mini_task=True)
+        yield Task("run")
```
Comment (JackTemaki) [inline g2p/apply.py:58]: If you have a regular job (no mini_task), you need to define a `self.rqmt` with some reasonable defaults. 
See https://github.com/rwth-i6/i6_core/blob/main/CONTRIBUTING.md (we should extend maybe the details in that document)

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: g2p/apply.py (hunk 1 above)
Comment (michelwi) [reply g2p/apply.py:58]: reasonable defaults can/should be task dependent (e.g. on the size of word list) or if that is not always realizable we can expose the rqmt to the user so he can in-/decrease them as needed.
In this case also make sure it does not enter the hash.

Comment (mattiadg) [reply g2p/apply.py:58]: Can the rqmt enter the hash?
And isn't the rqmt always exposed since it is a public attribute of Job?

Comment (JackTemaki) [reply g2p/apply.py:58]: In the usual way that there is only `self.rqmt = { some defaults }` if will not enter the hash and will be exposed for outside adjustment. I think Willi referred to the case that some jobs that often change requirements have them as actual constructor parameter, and then you need to make sure it is not part of the hash.

**Review by curufinwe — CHANGES_REQUESTED** [review]
Agree with @JackTemaki and @michelwi

**Comment by mattiadg** [comment]
Is it ok like this?

**Comment by curufinwe** [comment]
You still have to pass the requirements to the run Task.

File: g2p/apply.py (hunk 1 above)
Comment (michelwi) [reply g2p/apply.py:60]: To use the rqmt for the job:
```suggestion
        yield Task("run", rqmt=self.rqmt)
```

**Review by michelwi — CHANGES_REQUESTED** [review]
If you think 2h are enough for all/most cases, we can keep the defined rqmt's like this.

**Review by curufinwe — APPROVED** [review]

**Review by michelwi — APPROVED** [review]

**Review by JackTemaki — APPROVED** [review]

**Pull request merged**